File: passing_cars.py
def solution(A):
    passing_cars = 0
    previous_passing_cars = 0
    is_passing = False
    
    for index in range(0, len(A)):
        if A[index] == 0:
            is_passing = True
            previous_passing_cars += 1
            continue
        
        if A[index] == 1 and is_passing:
            passing_cars += 1*previous_passing_cars
        else:
            is_passing = False
            
        if passing_cars > 1000000000:
            return -1
            
    return passing_cars

File: PassingCars.cs
using System;

class Solution {
    public int solution(int[] A) {
        bool isPassing = false;
        int passingCars = 0;
        int previousPassingCars = 0;
       
        for (int index = 0; index < A.Length; index++)
        {
            if (A[index] == 0)
            {
                previousPassingCars++;   
                isPassing = true;
                continue;
            }
            
            if (A[index] == 1 && isPassing)
                   passingCars += 1*previousPassingCars;
            else 
                isPassing = false;
            
            if (passingCars  > 1000000000)
                return -1;
        }
        
        return passingCars;
    }
}